modules/optimizing-the-vllm-runtime.adoc

```diff
@@ -0,0 +1,169 @@
+      - --distributed-executor-backend=mp
+      - --speculative-model=/mnt/models/<path_to_speculative_model>
+      - --num-speculative-tokens=[NUM_SPECULATIVE_TOKENS]
+      - --use-v2-block-manager
+----
++ 
+Replace the `[NUM_SPECULATIVE_TOKENS]` with the desired value and `<path_to_speculative_model>` with the path to the speculative model on your S3-compatible object storage.
+. To configure the vLLM model-serving runtime for multi-modal inferencing, add the following arguments as shown:
++
+[source]
+----
+containers:
+  - args:
+      - --trust-remote-code
+----
++
+[NOTE]
+====
+Ensure that you use the `--trust-remote-code` argument with models from trusted sources. 
+====
+. Click Update.
++
+The *Serving runtimes* page opens and shows the list of runtimes that are installed. Observe that the custom model-serving runtime you updated is shown.
+. For speculative decoding, you must additionally redeploy the `InferenceService` custom resource definition (CRD) for the vLLM model-serving runtime as follows:
+.. Log in to the OpenShift CLI.
+.. List the available inference services in your namespace.
++
+[source]
+----
+oc get -n <namespace> isvc
+----
+Note the name of the `InferenceService` that needs to be re-deployed.
+.. Save the `InferenceService` manifest to a YAML file:
++
+[source]
+----
+oc get -n <namespace> isvc <inference-service-name> -o yaml > inferenceservice.yml
+----
+Replace the placeholder values with your own.
+.. Delete the exisiting `InferenceService`:
++
+[source]
+----
+oc delete -n <namespace> isvc <inference-service-name>
+----
+Replace the placeholder values with your own.
+.. Deploy the modified `InferenceService` CRD using the YAML file that you saved.
++
+[source]
+----
+oc apply -f inferenceservice.yml
+----
+.. Optional: Check the status of the `InferenceService` deployment as follows:
++
+[source]
+----
+oc get pod,isvc -n <namespace>
+----
+Replace placeholder values with your own.
+ifdef::upstream[]
+. Deploy the model by using the custom runtime  as described in {odhdocshome}{default-url}/serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Deploying models on the single-model serving platform].
+endif::[]
+ifndef::upstream[]
+. Deploy the model by using the custom runtime  as described in {rhoaidocshome}{default-url}/serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Deploying models on the single-model serving platform].
+endif::[]
+
+.Verification
+
+* If you have configured the vLLM model-serving runtime for speculative decoding, use the following example command to verify API requests to your deployed model:
++
+[source]
+----
+curl -v https://<inference_endpoint_url>:443/v1/chat/completions 
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <token>"
```

Comment:
Might be worth it to mention that the authorization header is only required if adding `--api-key` in the vllm commandline arguments (`InferenceService` or `ServingRuntime`):

```
--api-key API_KEY     If provided, the server will require this key to be
                        presented in the header.
```

Reply:
Not really. This authorization is intended for Authernio token, not for the vLLM API key. We don’t provide documentation for the API key on our VLLM server, so I don’t think it needs to be included here api key in our doc and have the doc as it is.